File: src/backend/ManagementHub.Service/Areas/Tests/TestViewModel.cs
using ManagementHub.Models.Domain.Language;
using ManagementHub.Models.Domain.Tests;

namespace ManagementHub.Service.Areas.Tests;

public class TestViewModel
{
	/// <summary>
	/// Title of the test (how it's displayed to users).
	/// </summary>
	public required string Title { get; set; }

	/// <summary>
	/// Block of text displayed to the refere before taking the test.
	/// </summary>
	public required string Description { get; set; }

	/// <summary>
	/// Language of the test.
	/// </summary>
	public required LanguageIdentifier Language { get; set; }

	/// <summary>
	/// The certification awarded for passing the test.
	/// </summary>
	public required Certification AwardedCertification { get; set; }

	/// <summary>
	///	Time limit in minutes.
	/// </summary>
	public required int TimeLimit { get; set; }

	/// <summary>
	/// Pass percentage.
	/// </summary>
	public required int PassPercentage { get; set; } = 80;

	/// <summary>
	/// How many questions to given to the referee during the test.
	/// </summary>
	public required int QuestionsCount { get; set; }

	/// <summary>
	/// If it's a recertification test for the previous rulebook.
	/// </summary>
	public bool Recertification { get; set; }

	/// <summary>
	/// Feedback to be displayed to the referee after the test if the pass.
	/// </summary>
	public string? PositiveFeedback { get; set; }

	/// <summary>
	/// Feedback to be displayed to the referee after the test if they fail.
	/// </summary>
	public string? NegativeFeedback { get; set; }

	/// <summary>
	/// Whether the test is active for all users.
	/// </summary>
	public bool Active { get; set; } = false;

	/// <summary>
	/// Identifier of the test.
	/// </summary>
	public TestIdentifier TestId { get; set; }
}

File: src/frontend/app/factories/testViewModel.ts
import { faker } from "@faker-js/faker";
import { Factory } from "fishery";

import { TestViewModel, CertificationLevel, CertificationVersion } from "../store/serviceApi";

export default Factory.define<TestViewModel>(
  ({ sequence }): TestViewModel => ({
    testId: sequence.toString(),
    title: `${faker.person.jobType()} Test`,
    description: faker.lorem.sentence(),
    language: "en-US",
    awardedCertification: {
      level: "assistant" as CertificationLevel,
      version: "twentyfour" as CertificationVersion,
    },
    timeLimit: 18,
    passPercentage: 80,
    questionsCount: 10,
    recertification: false,
    positiveFeedback: faker.lorem.sentence(),
    negativeFeedback: faker.lorem.sentence(),
    active: true,
  })
);
